### PR TITLE
fix(Button): support flip for cirulcar buttons

### PR DESCRIPTION
**Pull request by mintyfloss**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [ ] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

My code does...

### CODE REVIEW

**Comment by mintyfloss** [comment]
I have a snapshot to update, but I'm not able to push any changes :(

```
remote: error: GH006: Protected branch update failed for refs/heads/circular-flip.
remote: error: Required status check "continuous-integration/travis-ci" is expected. At least 1 approving review is required by reviewers with write access.
To github.com:RetailMeNot/anchor.git
 ! [remote rejected] circular-flip -> circular-flip (protected branch hook declined)
error: failed to push some refs to 'git@github.com:RetailMeNot/anchor.git'
```

File: src/Button/Flip/Flip.component.tsx [hunk 1]
```diff
@@ -1,25 +1,36 @@
...
     top: -1px;
     right: -1px;
-    width: 1.25rem;
-    height: 1.25rem;
     content: '';
 
+    ${({ circular }) =>
+        circular
+            ? css`
+                  border-top-right-radius: 1000px;
```
Comment (brandonlilly) [inline src/Button/Flip/Flip.component.tsx:22]: if `base` works below, then `circular` should work here.

Comment (mintyfloss) [reply src/Button/Flip/Flip.component.tsx:22]: Ah yes, I didn't realize `circular` was in the theme. It looks like I have more work to do in this PR. Circular flip only works in the `md` size. The problem is that I want to make the flip be 50% of the button height and make it a square. I haven't thought of a good way to do that yet.

Comment (brandonlilly) [reply src/Button/Flip/Flip.component.tsx:22]: make it a square on a circular button? do you have a mock?

`height` is something you can pass into the flip component from the button component, then just `calc(${height} / 2)` should work I think

**Pull request closed without merging**